cli/wallet/file/file.go

```diff
@@ -0,0 +1,348 @@
+package file
+
+import (
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/mapstructure"
+	flag "github.com/spf13/pflag"
+	bip39 "github.com/tyler-smith/go-bip39"
+	"golang.org/x/crypto/pbkdf2"
+
+	"github.com/oasisprotocol/deoxysii"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/sakg"
+	coreSignature "github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+
+	"github.com/oasisprotocol/oasis-sdk/cli/config"
+	"github.com/oasisprotocol/oasis-sdk/cli/wallet"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature/ed25519"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+const (
+	walletKind = "file"
+
+	cfgAlgorithm = "algorithm"
+	cfgNumber    = "number"
+
+	algorithmEd25519    = "ed25519"
+	algorithmEd25519Raw = "ed25519-raw"
+
+	stateKeySize  = 32
+	stateSaltSize = 32
+)
+
+type walletConfig struct {
+	Algorithm string `mapstructure:"algorithm"`
+	Number    uint32 `mapstructure:"number,omitempty"`
```

Comment:
I think this is never assigned. how do you envision this being populated in the future? fed in through a new field in ImportSource maybe?